### PR TITLE
[MD] Change cluster selector component name to data source selector

### PR DESCRIPTION
**Pull request by BionIT**
### Description
This change renames clusterselector to datasource selector 

### Issues Resolved
Fixes https://github.com/opensearch-project/OpenSearch-Dashboards/issues/6038

## Screenshot

https://github.com/opensearch-project/OpenSearch-Dashboards/assets/17714150/e7767763-1db2-442f-a9ac-0f3b961c64ec

## Testing the changes
The following steps were performed in the recording above:
1. enable data source plugin
2. add sample data should render data source selector correctly and can select data source and add sample data, and sample data should render correctly
3. in dev tools, select a data source and perform query should render results correctly
4. go to query workbench, should render data source selector correctly and can select data source and request to yield indices
5. go to alert plugin, should render data source selector correctly and can select data source and request to yield monitors from data source

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by xinruiba — APPROVED** [review]

**Review by yujin-emma — APPROVED** [review]

**Review by seraphjiang — APPROVED** [review]
Awesome, thanks

**Comment by ashwin-pc** [comment]
Isnt Data Connection what we aligned on for the name here? https://github.com/opensearch-project/OpenSearch-Dashboards/issues/5790#issuecomment-1930916400

**Comment by BionIT** [comment]
> Isnt Data Connection what we aligned on for the name here? [#5790 (comment)](https://github.com/opensearch-project/OpenSearch-Dashboards/issues/5790#issuecomment-1930916400)

This is the current proposal https://github.com/opensearch-project/OpenSearch-Dashboards/issues/6039

**Comment by ashwin-pc** [comment]
That issue was opened 5 hours ago though. Let me take the conversation there

**Comment by seraphjiang** [comment]
> That issue was opened 5 hours ago though. Let me take the conversation there. Can we give people time to respond to the proposal?

Thanks @ashwin-pc  replied in rfc, 

It is expecting to disambious this in 2.13 for developer. It makes sense to give people time for feedback.

**Review by ZilongX — APPROVED** [review]

File: src/plugins/data_source_management/public/components/cluster_selector/create_cluster_selector.tsx [hunk 1]
```diff
@@ -1,11 +0,0 @@
-/*
```
Comment (Flyingliuhub) [inline src/plugins/data_source_management/public/components/cluster_selector/create_cluster_selector.tsx:1]: do we need to change file name as well? the file name still using the cluster_selector?

**Review by Flyingliuhub — APPROVED** [review]

File: src/plugins/data_source_management/public/components/cluster_selector/create_cluster_selector.tsx (hunk 1 above)
Comment (BionIT) [reply src/plugins/data_source_management/public/components/cluster_selector/create_cluster_selector.tsx:1]: yeah, the folder has been removed

**Pull request merged**